src/functions/Get-SkipRemainingOnFailurePlugin.ps1

```diff
@@ -1,21 +1,46 @@
-function New-SkippedTestMessage {
-    [OutputType([string])]
-    [CmdletBinding()]
-    param (
-        [Parameter(Mandatory)]
-        [Pester.Test]
-        $Test
-    )
-    "Skipped due to previous failure at '$($Test.ExpandedPath)' and Run.SkipRemainingOnFailure set to '$($PesterPreference.Run.SkipRemainingOnFailure.Value)'"
-}
-
 function Resolve-SkipRemainingOnFailureConfiguration {
     $supportedValues = 'None', 'Block', 'Container', 'Run'
     if ($PesterPreference.Run.SkipRemainingOnFailure.Value -notin $supportedValues) {
         throw (Get-StringOptionErrorMessage -OptionPath 'Run.SkipRemainingOnFailure' -SupportedValues $supportedValues -Value $PesterPreference.Run.SkipRemainingOnFailure.Value)
     }
 }
 
+function Set-RemainingAsSkipped {
+    param(
+        [Parameter(Mandatory)]
+        [Pester.Test]
+        $FailedTest,
+
+        [Parameter(Mandatory)]
+        [Pester.Block]
+        $Block
+    )
+
+    $errorRecord = [Pester.Factory]::CreateErrorRecord(
+        'PesterTestSkipped',
+        "Skipped due to previous failure at '$($FailedTest.ExpandedPath)' and Run.SkipRemainingOnFailure set to '$($PesterPreference.Run.SkipRemainingOnFailure.Value)'",
+        $null,
+        $null,
+        $null,
+        $false
+    )
+
+    Fold-Block -Block $Block -OnTest {
+        param ($test)
+        if ($test.ShouldRun -and -not $test.Skip -and -not $test.Executed) {
+            # Skipping and counting remaining unexecuted tests
+            $Context.Configuration.SkipRemainingOnFailureCount += 1
+            $test.Skip = $true
+            $test.ErrorRecord.Add($errorRecord)
+        }
+    } -OnBlock {
+        param($block)
+        if ($block.ShouldRun -and -not $block.Skip -and -not $block.Executed) {
+            # Marking remaining blocks as Skip to avoid executing BeforeAll/AfterAll
+            $block.Skip = $true
+        }
```

Comment:
Should the block also get a ErrorRecord to explain why it was skipped?

Reply:
Not sure, do we write them all out?

Reply:
Then you probably don't care if there are no tests, and if there are tests you will see the skipped tests, so imho adding the error is not necessary?